keyform/models.py

```diff
@@ -29,25 +51,35 @@ class Request(models.Model):
         ('ch', _('Check')),
     )
 
-    STATUS_TYPES = (
-        ('pr', _('Processing')),
-        ('ks', _('Key Sent')),
-        ('kd', _('Key Distributed')),
-    )
-
-    bpn_validator = RegexValidator('[mM8]\d{8}', "Bearpass number must start with an 'M,' 'm,' or '8,' and followed by eight digits.'")
+    bpn_validator = RegexValidator(
+        '[mM8]\d{8}',
+        "Bearpass number must start with an 'M,' 'm,' or '8,' and followed by eight digits.'")
 
     building = models.ForeignKey(Building)
     student_name = models.CharField(max_length=128, blank=True)
     reason_for_request = models.CharField(max_length=2, choices=REQUEST_TYPES)
-    amt_received = models.DecimalField(max_digits=7, decimal_places=2, default=0, blank=True, verbose_name= _('Amount received'), validators=[MinValueValidator(Decimal('0.00'))])
+    amt_received = models.DecimalField(
+        max_digits=7,
+        decimal_places=2,
+        default=0,
+        blank=True,
+        verbose_name=_('Amount received'),
+        validators=[
+            MinValueValidator(
+                Decimal('0.00'))])
     payment_method = models.CharField(max_length=2, choices=PAYMENT_TYPES, null=True, blank=True)
-    charge_amount = models.DecimalField(max_digits=7, decimal_places=2, validators=[MinValueValidator(Decimal('0.00'))])
+    charge_amount = models.DecimalField(
+        max_digits=7, decimal_places=2, validators=[
+            MinValueValidator(
+                Decimal('0.00'))])
     staff = models.ForeignKey(settings.AUTH_USER_MODEL)
     bpn = models.CharField(max_length=9, verbose_name=_('M-Number'), validators=[bpn_validator])
     created_timestamp = models.DateTimeField(default=now, blank=True)
     charged_on_rcr = models.BooleanField(default=False, verbose_name=_('Charged on RCR'))
-    status = models.CharField(max_length=2, choices=STATUS_TYPES, default = 'pr')
+    status = models.ForeignKey(Status, related_name="status", default=Status.objects.all()[0])
+    previous_status = models.ForeignKey(Status, related_name="previous_status", default=Status.objects.all()[0])
```

Comment:
What if we go ahead and include this in the create form, but just autofill it to processing?